test/lib.sh

```diff
@@ -44,6 +44,35 @@ empty_repo() {
     echo $tmpd
 }
 
+## bootstrap_project PATH TEST_PROJECT INITIAL_CONVENTION
+#
+# Build a temp boilerplated git project containing test_project files
+# - Copies in boilerplate/update from $REPO_ROOT
+# - Creates an empty boilerplate/update.cfg
+# It DOES run the update in order to allow using boilerplate/generated-includes.mk
+#
+# :param PATH: An existing directory that has been `git init`ed, like
+#       what you get when you run `empty_repo`.
+# :pranm TEST_PROJECT: the test_project (from boilerplate/test/test_projects)
+# :param INITIAL_CONVENTION: The convention(s) to be used for initializing the 
+#       project
+bootstrap_project() {
+    repodir=$1
+    test_project=$2
+    (
+        cp -R $REPO_ROOT/test/projects/$test_project/* $repodir/.
+        pushd $repodir
```

Comment:
nit: since you're running this chunk in a subshell (the parens on L62 & 73) you could just 
```suggestion
        cd $repodir
```
here, and...